homeassistant/components/remootio/__init__.py

```diff
@@ -49,4 +49,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
+
+    if DOMAIN in hass.data.keys():
+        hass.data[DOMAIN].pop(entry.entry_id, None)
+
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
```

Comment:
Wrong order here. You need to unload platforms first . Verify that it succeeded, then terminate client and pop the data.

Reply:
Adapted.